stdlib/public/core/StringCreate.swift

```diff
@@ -50,18 +50,51 @@ internal func _allASCII(_ input: UnsafeBufferPointer<UInt8>) -> Bool {
 }
 
 extension String {
-  @usableFromInline
-  internal static func _fromASCII(
-    _ input: UnsafeBufferPointer<UInt8>
-  ) -> String {
+  
+  internal static func _uncheckedFromASCII(_ input: UnsafeBufferPointer<UInt8>)
+  -> String {
     _internalInvariant(_allASCII(input), "not actually ASCII")
 
     if let smol = _SmallString(input) {
       return String(_StringGuts(smol))
     }
-
+    
+    return __StringStorage.create(
+      initializingFrom: input, isASCII: true
+    ).asString
+  }
+  
+  internal static func _fromASCIIRepairing(
+    _ input: UnsafeBufferPointer<UInt8>
+  ) -> (result: String, repairsMade: Bool) {
+    
+    if let smol = _SmallString(input) {
+      return (String(_StringGuts(smol)), false)
+    }
+    
     let storage = __StringStorage.create(initializingFrom: input, isASCII: true)
```

Comment:
I'm surprised we don't have an invariant check for this. You can alternatively set isASCII at the end of the function.